### PR TITLE
Add more detailed Forc.toml manifest reference documentation

### PR DESCRIPTION
**Pull request by kayagokalp**
`Manifest Reference` section added to The Sway Book `forc` section.

### CODE REVIEW

File: docs/src/forc/manifest_reference.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+The `Forc.toml` (the *manifest* file) is a compulsory file for each package and it is written in [TOML] format. `Forc.toml` consists of the following fields:
+
+* [`[project]`](#the-project-section) — Defines a sway project.
+  * [`name`](#) — The name of the project.
+  * [`authors`](#) — The authors of the project.
+  * [`organization`](#) — The organization of the project.
+  * [`license`](#) — The project license.
+  * [`entry`](#) — The entry point of the project.
+  * [`implicit_std`](#) -  Controls whether provided `std` version (with current `forc` version) will get added as a dependency *implicitly*.
```
Comment (adlerjohn) [inline docs/src/forc/manifest_reference.md:11]: `_` for italics. Also, install mdlint for your IDE to auto-format and fix errors like extra or missing spacing.

```suggestion
  * [`implicit_std`](#) -  Controls whether provided `std` version (with current `forc` version) will get added as a dependency _implicitly_.
```

File: docs/src/forc/manifest_reference.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Manifest Reference
+
+The `Forc.toml` (the _manifest_ file) is a compulsory file for each package and it is written in [TOML] format. `Forc.toml` consists of the following fields:
+
+* [`[project]`](#the-project-section) — Defines a sway project.
+  * `name` — The name of the project.
+  * `authors` — The authors of the project.
+  * `organization` — The organization of the project.
+  * `license`— The project license.
+  * `entry` — The entry point of the project.
```
Comment (adlerjohn) [inline docs/src/forc/manifest_reference.md:10]: Entry point is actually non-obvious, so should be documented more.

File: docs/src/forc/manifest_reference.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  * `implicit_std` -  Controls whether provided `std` version (with current `forc` version) will get added as a dependency _implicitly_.
```
Comment (adlerjohn) [inline docs/src/forc/manifest_reference.md:11]: Add a recommendation that this should be left as default unless you really know what you're doing.

File: docs/src/forc/manifest_reference.md [hunk 4]
```diff
@@ -8,7 +8,10 @@ The `Forc.toml` (the _manifest_ file) is a compulsory file for each package and
   * `organization` — The organization of the project.
   * `license`— The project license.
   * `entry` — The entry point of the project.
-  * `implicit_std` -  Controls whether provided `std` version (with current `forc` version) will get added as a dependency _implicitly_.
+    * For `scripts` and `predicates` point to the file with `main` function.
+    * For `libraries` see [Libraries](./../sway-program-types/libraries.md).
+    * For `contracts` point to the file with the `ABI` implementation.
```
Comment (kayagokalp) [inline docs/src/forc/manifest_reference.md:13]: Not really sure about the `contracts` part.

**Review by adlerjohn — APPROVED** [review]
I'm okay with this as a start. Eventually it should be made to more closely match https://doc.rust-lang.org/cargo/reference/manifest.html, but that can be done in follow-up PRs.

**Pull request merged**